rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -79,11 +99,71 @@ public BakeRecipe buildBakeRecipe(
     command.add("kustomize");
     command.add("build");
     command.add(templatePath.getParent().toString());
+
+    BakeRecipe result = new BakeRecipe();
     result.setCommand(command);
+    return result;
+  }
+
+  private BakeRecipe buildBakeRecipeFromGitRepo(
+      BakeManifestEnvironment env, KustomizeBakeManifestRequest request, Artifact artifact)
+      throws IOException {
+    // This is a redundant check for now, but it's here for when we soon remove the old logic of
+    // building from
+    // a github/file artifact type and instead, only support the git/repo artifact type
+    if (!"git/repo".equals(artifact.getType())) {
+      throw new IllegalArgumentException("The inputArtifact should be of type \"git/repo\".");
+    }
+
+    String kustomizeFilePath = request.getKustomizeFilePath();
+    if (kustomizeFilePath == null) {
+      throw new IllegalArgumentException("The bake request should contain a kustomize file path.");
+    }
+
+    Path tarPath = env.resolvePath("repo.tar.gz");
+    try {
+      artifactDownloader.downloadArtifact(artifact, tarPath);
+    } catch (IOException e) {
+      throw new IOException("Failed to download git/repo artifact: " + e.getMessage(), e);
+    }
 
+    try {
+      extractArtifact(tarPath, env.resolvePath(""));
+    } catch (IOException e) {
+      throw new IOException("Failed to extract git/repo artifact: " + e.getMessage(), e);
+    }
+
+    List<String> command = new ArrayList<>();
+    command.add("kustomize");
+    command.add("build");
+    command.add(env.resolvePath(kustomizeFilePath).getParent().toString());
+
+    BakeRecipe result = new BakeRecipe();
+    result.setCommand(command);
     return result;
   }
 
+  // This being here is temporary until we find a better way to abstract it
+  public static void extractArtifact(Path inputTar, Path outputPath) throws IOException {
+    TarArchiveInputStream tarArchiveInputStream =
```

Comment:
This stream won't get closed if there's an exception in the logic below; this can be fixed by using a try-with-resources construct:
```
try (TarArchiveInputStream tarArchiveInputStream = new TarArchiveInputStream(
            new GzipCompressorInputStream(new BufferedInputStream(Files.newInputStream(inputTar)))) {
//...
}
```